### PR TITLE
Button board

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/oxon/buttonboard/default.schema.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      "title": "Message type",
+      "type": "integer",
+      "description": "Type of message",
+      "minimum": 0,
+      "maximum": 255
+    },
+    "buttonClicked": {
+      "title": "Button Clicked",
+      "type": "integer",
+      "description": "Defines if and which button got pressed. If the output is 0 no button got pressed",
```
Comment (mariusgiger) [inline types/oxon/buttonboard/default.schema.json:17]: got -> is

File: types/oxon/buttonboard/default.schema.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      "minimum": 0,
+      "maximum": 255
+    },
+    "imageID": {
+      "title": "Image ID",
+      "type": "integer",
+      "description": "The ID of the Image used when the button sent the uplink",
```
Comment (mariusgiger) [inline types/oxon/buttonboard/default.schema.json:24]: image?

File: types/oxon/buttonboard/lifecycle.schema.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "properties": {
+    "statusPercent": {
+      "type": "number",
+      "minimum": 0,
+      "maximum": 100,
+      "title": "Battery charge",
+      "description": "The battery charge in percent.",
+      "unit": "%"
+    },
+    "heartbeat": {
```
Comment (mariusgiger) [inline types/oxon/buttonboard/lifecycle.schema.json:15]: how can this be false?

File: types/oxon/buttonboard/uplink.spec.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          assert.equal(value.data.buttonClicked, 0);
+          assert.equal(value.data.imageH, 1);
+          assert.equal(value.data.imageID, 63);
+          assert.equal(value.data.temperature, 18);
+
+          validate(value.data, defaultSchema, { throwError: true });
+        }
+      });
+      consume(data);
+      done();
```
Comment (mariusgiger) [inline types/oxon/buttonboard/uplink.spec.js:78]: remove done

File: types/oxon/buttonboard/uplink.spec.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+          assert.equal(value.data.msgType, 49);
+          assert.equal(value.data.buttonClicked, 0);
+          assert.equal(value.data.imageH, 1);
+          assert.equal(value.data.imageID, 63);
+          assert.equal(value.data.temperature, 18);
+
+          validate(value.data, defaultSchema, { throwError: true });
+        }
+      });
+      consume(data);
```
Comment (mariusgiger) [inline types/oxon/buttonboard/uplink.spec.js:77]: validate that a sample for default and lifecycle has been emitted

**Review by mariusgiger — CHANGES_REQUESTED** [review]

File: types/oxon/buttonboard/lifecycle.schema.json (hunk 3 above)
Comment (J2thatsme) [reply types/oxon/buttonboard/lifecycle.schema.json:15]: If its 0 so a keepalive; will clarify this

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**